### ISSUE TITLE
feat: support tsconfig.json  -> `"exclude": ["**/*.{test,spec}.ts"]`

### ISSUE THREAD

**Issue by aladdin-add** [microsoft/TypeScript#47001]
# Suggestion

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [ x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [ x] This wouldn't change the runtime behavior of existing JavaScript code
* [ x] This could be implemented without emitting different JS based on the types of the expressions
* [ x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [ x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

a `tsconfig.json` example: 
```json
{
  "extends": "@tsconfig/node16/tsconfig.json",
  "include": ["lib/**/*.ts"],
  "exclude": ["**/*.{test,spec}.ts"]
}
```

`"exclude": ["**/*.{test,spec}.ts"]` does not work as expected, and *no* warnings/errors output (ts v4.5.2).
I'd propose support patterns like this, as it's very widely used in js community, e.g. babel/jest/eslint/....

**Comment by MartinJohns** [comment]
But that exists already: https://www.typescriptlang.org/tsconfig#exclude

**Comment by aladdin-add** [comment]
yes, it supports `"exclude": ["**/*.test.ts"]`, but not `"exclude": ["**/*.{test,spec}.ts"]` (the glob patterns)

**Comment by MartinJohns** [comment]
Perhaps you should make more clear that you want more advanced glob patterns supported.

For the mentioned example you can just use: `["**/*.spec.ts", "**/*.test.ts"]`

**Comment by eng-dibo** [comment]
voting for this feature
use case: decrease the length if the command if used in an npm script, i.e not 8n tsconfig.json file

**Comment by shellscape** [comment]
Here from the future to vote for this feature to reduce the silly size of `exclude` and `include` in large projects.
